code/datums/abilities/flock/flockmind.dm

```diff
@@ -477,10 +477,30 @@
 /datum/targetable/flockmindAbility/droneControl
 	cooldown = 0
 	icon = null
-	var/task_type
 	var/mob/living/critter/flock/drone/drone = null
 
 /datum/targetable/flockmindAbility/droneControl/cast(atom/target)
+	var/datum/component/flock_ping/selected/ping = drone.GetComponent(/datum/component/flock_ping/selected)
+	ping.RemoveComponent()
+	qdel(ping)
+	if (target == src.drone)
+		return
+	//by default we just walk somewhere
+	var/task_type = /datum/aiTask/sequence/goalbased/flock/build/targetable
+	if (ismob(target) && !isflockmob(target))
```

Comment:
Not sure how much it matters, but means you can't target other Flocks, though I suppose that's kind of broken atm rn

Also, could check for more than just mobs? There's a list of stuff checked for in /datum/aiTask/timed/targeted/flockdrone_shoot/targetable/switched_to() that could be used, would allow for more targeting than just mobs